### PR TITLE
[tools\depends] Reduce greying of developer hair

### PR DESCRIPTION
**Pull request by garbear**
## Description

One of the largest sources of rebase conflicts is with the two depends Makefiles, causing endless frustration for frequent rebasers. Let's alphabetize these and save us the grey hair.

## Motivation and context

And after making this change I went to rebase it, and hit this!

```
Auto-merging tools/depends/target/Makefile
Auto-merging tools/depends/native/Makefile
CONFLICT (content): Merge conflict in tools/depends/native/Makefile
error: could not apply a280e4d06b... Reduce greying of developer hair
Could not apply a280e4d06b... Reduce greying of developer hair
```

So grey hair not avoided for me.

## How has this been tested?

Tested by building depends via `make`.

## What is the effect on users?

* Happier developers

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

### CODE REVIEW

File: tools/depends/native/Makefile [hunk 1]
```diff
@@ -38,36 +55,37 @@ all: native
...
-autoconf-archive: autoconf
 dpkg: automake gettext libtool pkg-config tar
 flatbuffers: cmake
 heimdal: libtool
-libtool: automake
+JsonSchemaBuilder: automake
 libjpeg-turbo: cmake nasm
 libpng: zlib automake
+libtool: automake
+libtool: automake
```
Comment (fuzzard) [inline tools/depends/native/Makefile:68]: Duplicate

Comment (garbear) [reply tools/depends/native/Makefile]: haha I told you I hit a rebase with a conflict. clear evidence that this change is needed

**Review by fuzzard — APPROVED** [review]
I'm ok for the change. Now I need to go merge all my open PRS so I don't have to rebase first! Haha

**Review by lrusak — APPROVED** [review]
I didn't check every line but I did a spot check on more than a few. Looks good. It's about time we alphabetize!

**Comment by garbear** [comment]
I'm ready to count the number of conflicts Jenkins is about to drop

File: tools/depends/native/Makefile [hunk 2]
```diff
@@ -4,15 +4,32 @@ ifneq ($(shell test -f $(NATIVEPREFIX)/share/config.site && echo 1),1)
...
+  autoconf-archive \
+  automake \
+  cmake \
+  flatbuffers \
+  gas-preprocessor \
+  gettext \
+  heimdal \
+  JsonSchemaBuilder \
+  libjpeg-turbo \
+  liblzo2 giflib \
```
Comment (Rechi) [inline tools/depends/native/Makefile:18]:
```suggestion
  liblzo2 \
  giflib \
```

File: tools/depends/target/Makefile [hunk 3]
```diff
@@ -5,17 +5,59 @@ ifneq ($(shell test -f $(PREFIX)/share/config.site && echo 1),1)
...
+  libfmt \
+  libfstrcmp \
+  libgcrypt \
+  libgpg-error \
+  libjpeg-turbo \
+  liblzo2 \
+  libmicrohttpd \
+  libnfs \
+  libplist \
+  libpng fribidi \
```
Comment (Rechi) [inline tools/depends/target/Makefile:39]:
```suggestion
  libpng \
  fribidi \
```

File: tools/depends/target/Makefile [hunk 4]
```diff
@@ -5,17 +5,59 @@ ifneq ($(shell test -f $(PREFIX)/share/config.site && echo 1),1)
...
+  libshairplay \
+  libspdlog \
+  libudfread \
+  libusb \
+  libxml2 \
+  libxslt  \
```
Comment (Rechi) [inline tools/depends/target/Makefile:45]:
```suggestion
  libxslt \
```

Comment (fuzzard) [reply tools/depends/target/Makefile:45]: Just as an fyi @Rechi im looking to piggy back these suggestions in with https://github.com/xbmc/xbmc/pull/20381

Comment (garbear) [reply tools/depends/target/Makefile:45]: thanks for cleaning up

**Pull request merged**